openedx_learning/apps/core/itemstore/models.py

```diff
@@ -0,0 +1,133 @@
+"""
+The item model hiearachy is: Item -> ItemVersion -> ItemInfo -> ItemRaw
+
+Item is the versionless thing that is guaranteed to exist for the lifetime of
+the LearningContext. An ItemVersion is a different version of that item for a
+given LearningContext, and may include policy changes (like grading). ItemInfo
+covers basic metadata that is intrinsic to the item itself, and now how it's
+used in a LearningContext. ItemRaw represents the raw byte data.
+
+TODO: Add link to ADR after it merges.
+"""
+from django.db import models
+
+from openedx_learning.lib.fields import (
+    hash_field,
+    identifier_field,
+    immutable_uuid_field,
+    manual_date_time_field,
+)
+from ..publish.models import LearningContext, LearningContextVersion
+
+
+class Item(models.Model):
+    """
+    This represents any content that has ever existed in a LearningContext.
+    """
+    uuid = immutable_uuid_field()
+    learning_context = models.ForeignKey(LearningContext, on_delete=models.CASCADE)
```

Comment:
> Is this the LearningContext that the Item is being used in, or the LearningContext that it's defined in (eg, a content library)? Is that distinction something that even exists at this level?

The one it's being used in, because Item has the identifier, and we need to be able to hang course-level policy changes off of ItemVersion. So the hierarchy kinda looks kind of like this:

```mermaid
graph TD;
    CI[Course Item]-->CIV[Course ItemVersion];
    CIV-->LII[Library ItemInfo];
    LII-->LIR[Library ItemRaw];
    LI[Library Item]-->LIV[Library ItemVersion];
    LIV-->LII;
```

So the course makes a copy of the high level objects (`Item`, `ItemVersion`), and shares the lower-level ones (`ItemInfo`, `ItemRaw`, and anything derived from those).

Reply:
That makes sense! The graphic is immensely helpful. I have to admit that the names of the models on their own did not guide me to this understanding.

The split between [Item , ItemVersion] and [ItemInfo, ItemRaw] reminds me of the XBlock framework's distinction between Usage and Definition, respectively. Is that a fair parallel?